templates/stamp/utilities.sh

```diff
@@ -836,44 +849,34 @@ setup_backup()
     databaseType="${10}";                                   # Database Type : mysql|mongo
 
     databaseUser="${11}"; databasePassword="${12}";         # Credentials for accessing the database for backup purposes
-    tempDatabaseUser="${13}"; tempDatabasePassword="${14}"; # Temporary credentials for accessing the backup (optional)
+    backupLocalPath="${13}";                                # Database Type : mysql|mongo
+    # Optional.
+    tempDatabaseUser="${14}"; tempDatabasePassword="${15}"; # Temporary credentials for accessing the backup (optional)
 
     log "Setting up database backup for '${databaseType}' database(s)"
 
-    # For simplicity, we require all parameters are set
-    if [ "$#" -lt 12 ]; then
-        echo "Not all required backup configuration parameters have been set"
+    # For simplicity, we require all required parameters are set
+    if [ "$#" -lt 13 ]; then
+        echo "Some required backup configuration parameters are missing"
         exit 1;
     fi
 
     # persist the settings
-    tee "${backup_configuration}" > /dev/null <<EOF
-BACKUP_STORAGEACCOUNT_NAME={BACKUP_STORAGEACCOUNT_NAME}
-BACKUP_STORAGEACCOUNT_KEY={BACKUP_STORAGEACCOUNT_KEY}
-BACKUP_RETENTIONDAYS={BACKUP_RETENTIONDAYS}
-MONGO_REPLICASET_CONNECTIONSTRING={MONGO_REPLICASET_CONNECTIONSTRING}
-MYSQL_SERVER_LIST={MYSQL_SERVER_LIST}
-DATABASE_USER={DATABASE_USER}
-DATABASE_PASSWORD={DATABASE_PASSWORD}
-TEMP_DATABASE_USER={TEMP_DATABASE_USER}
-TEMP_DATABASE_PASSWORD={TEMP_DATABASE_PASSWORD}
-DATABASE_TYPE={DATABASE_TYPE}
-EOF
-
-    # replace the place holders (using # since the repo path will have forward slashes)
-    log "Populating backup configuration file with appropriate values"
-    sed -i "s#{BACKUP_STORAGEACCOUNT_NAME}#${account_name}#I" $backup_configuration
-    sed -i "s#{BACKUP_STORAGEACCOUNT_KEY}#${account_key}#I" $backup_configuration
-    sed -i "s#{BACKUP_RETENTIONDAYS}#${backupRententionDays}#I" $backup_configuration
-    sed -i "s#{MONGO_REPLICASET_CONNECTIONSTRING}#${mongoReplicaSetConnectionString}#I" $backup_configuration
-    sed -i "s#{MYSQL_SERVER_LIST}#${mysqlServerList}#I" $backup_configuration
-    sed -i "s#{DATABASE_USER}#${databaseUser}#I" $backup_configuration
-    sed -i "s#{DATABASE_PASSWORD}#${databasePassword}#I" $backup_configuration
-    sed -i "s#{TEMP_DATABASE_USER}#${tempDatabaseUser}#I" $backup_configuration
-    sed -i "s#{TEMP_DATABASE_PASSWORD}#${tempDatabasePassword}#I" $backup_configuration
-    sed -i "s#{DATABASE_TYPE}#${databaseType}#I" $backup_configuration
-
-    # this file contains secrets (storage account key). Secure it
+    bash -c "cat <<EOF >${backup_configuration}
+BACKUP_STORAGEACCOUNT_NAME=${account_name}
```

Comment:
i remember we had a little trouble with envsubst a while back, but this solution doesn't use that binary. in fact: this is so reliable that we've been using it in the three ARM Templates we contributed to the the QuickStart Template library last year.